lib/Backends/NNPI/NNPI.cpp

```diff
@@ -84,22 +88,22 @@ NNPIBackendOptions NNPIBackend::backendOptions_;
 NNPIAdapterContainer NNPIBackend::adapter_;
 
 unsigned NNPIBackend::numDevices() {
```

Comment:
@omromano @jsubag 
This change to numDevices reverts a change I made to your last PR for 5.1.4.
numDevices as it is coded will leak memory if run on a host that doesn't have a card attached (adapterCreate will allocate, adapterGetInfo will fail and we return without calling adapterDestroy). numDevices() on github master fixes this - please update your internal sources for that change so we don't try to revert it on the next PR.